Sources/MapboxCoreNavigation/Router.swift

```diff
@@ -351,6 +364,20 @@ extension InternalRouter where Self: Router {
                 return completion(session, .success(.init(routeResponse: response, routeIndex: mostSimilarIndex)))
             }
         }
+        
+        switch delegate?.router(self, requestSourceForReroutingWith: options) {
+        case .default, .none:
+            routeTask = routingProvider.calculateRoutes(options: options) {(session, result) in
+                parseResult(session, result)
+                self.routeTask = nil
+            }
+        case .custom(let reroutingResult):
+            let (options, result) = reroutingResult()
```

Comment:
Are we expecting that the user will alter the options for his request?

Reply:
Since user is able to provide his own response, it seems logical to give him control over altering options as well.